index.js

```diff
@@ -43,18 +43,29 @@ exports.tokenBefore = function(token) {
     }
 
     // we always normalize the behavior to remove unnecessary escapes
-    var alternateEscape = new RegExp('\\\\' + alternate, 'g');
-    content = content.replace(alternateEscape, alternate);
+    // If the escaped quote should be unescaped, then escape it (e.g. '\"' -> '"')
+    //   However, don't remove escapes from slashes (e.g. '\\"' -> '\\"')
+    // DEV: RegExp explained:
+    //   - We want to start on a non-slash character to only match at boundaries
+    //      (i.e. `^|\`, the 4 slashes are 1 slash in RegExp)
+    //   - We want to prevent replacing escaped slashes (i.e. `\\` should not be replaced, this is 8 slashes in RegExp)
+    //   - We want to replace `\"` with `"` as the slash is unnecessary
+    //      so consume `\` but don't match it and preserve the quote
+    ///  - We match as many pairs as possible for `\"` due to requiring a leading word boundary on our next match (which won't exist)
+    var alternateEscape = new RegExp('(^|[^\\\\])((\\\\{2})*)((\\\\' + alternate + ')+)', 'g');
+    content = content.replace(alternateEscape, function replaceQuotes (input, boundaryChar, leadingEscapedSlashes, leadingEscapedSlash, escapedQuoteChars, escapedQuoteChar, match) {
+      // DEV: We divide the escapedQuoteChars by 2 since there are 2 characters in each escaped part ('\\"'.length === 2)
+      return boundaryChar + leadingEscapedSlashes + new Array((escapedQuoteChars.length / 2) + 1).join(alternate);
```

Comment:
what about we extract the `new Array(len).join()` into a `repeat` function? that way it's a little bit clearer:

```js
function repeat(str, n) {
  return new Array(n + 1).join(str);
}
```